dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/CommonTaskProcessor.java

```diff
@@ -126,6 +133,14 @@ public boolean dispatchTask() {
                 return false;
             }
 
+            // check in advance to avoid invalid infinite loops in TaskPriorityQueueConsumer
+            if (CollectionUtils.isEmpty(serverNodeManager.getWorkerGroupNodes(taskExecutionContext.getWorkerGroup()))) {
```

Comment:
We cannot make this change easily, the workgroup may loss in some cases, and recovery automically.

And this is not safe, the workflow may exist in the processor but loss in dispatch part.

BTY, it's not suggested to do this validation master, we need to do the validation in api-side.

Reply:
> We cannot make this change easily, the workgroup may loss in some cases, and recovery automically.
> 
> And this is not safe, the workflow may exist in the processor but loss in dispatch part.
> 
> BTY, it's not suggested to do this validation master, we need to do the validation in api-side.

but if ds want to worker group recovery automically, we also should not add this validation in api-side,just keep loop waiting. WDYT @ruanwenjun

Reply:
Add validation in API-Side can stop the process in workflow creation.

Reply:
> Add validation in API-Side can stop the process in workflow creation. In fact, we have this check in front-end, but in api-side, we don't validate the workgroup.

u are correct, i add a validator in api side and remove it in master side